api/solarperformanceinsight_api/compute.py

```diff
@@ -254,9 +258,17 @@ def process_single_modelchain(
         )
     # mean
     weather_avg: pd.DataFrame = weather_sum / num_arrays  # type: ignore
-    adjusted_zenith: pd.DataFrame = adjust(
-        results.solar_position[["zenith"]]
-    )  # type: ignore
+    adjusted_zenith: pd.DataFrame
+    # not calculate if effective irradiance is provided
+    if results.solar_position is not None:
+        adjusted_zenith = adjust(results.solar_position[["zenith"]])  # type: ignore
+    else:
+        # calculate solar position making sure to shift times and shift back
+        adjusted_zenith = adjust(
+            mc.location.get_solarposition(
+                weather_avg.index.shift(freq=tshift)  # type: ignore
+            )[["zenith"]]
```

Comment:
maybe worth noting modelchain passes pressure and air temperature to get_solarposition, but there's no difference if we extract zenith rather than apparent zenith.

https://github.com/pvlib/pvlib-python/blob/b666520f84cee895daa3418e06d8d8a543fc1b38/pvlib/modelchain.py#L1223-L1239

@cwhanse a nuance to be aware of

Reply:
@wholmgren are you saying that there's no practical difference between `zenith` and `apparent_zenith` in this use? Because there should be a difference in the values, albeit quite small.

Reply:
no, I'm saying that because we use zenith we don't need to worry about inconsistencies between this explicit call to `get_solarposition` and the call within `ModelChain`. But that it would be a problem if we changed the SPI results to use apparent zenith rather than zenith.